src/frontend/planner_test/tests/testdata/batch_index_join.yaml

```diff
@@ -58,3 +58,18 @@
     └─BatchLookupJoin { type: Inner, predicate: t.a = idx.c, output: all }
       └─BatchExchange { order: [], dist: UpstreamHashShard(t.a) }
         └─BatchScan { table: t, columns: [t.a, t.b], distribution: SomeShard }
+- name: test index join distribution derive
+  sql: |
+    set rw_batch_enable_lookup_join = true;
+    set query_mode = distributed;
+    create table t (a int, b int);
+    create table t2 (c int, d int, primary key(c));
+    select t2.c, t2.d, count(distinct t.a) from t join t2 on t.a = t2.c group by t2.c, t2.d;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+    └─BatchHashAgg { group_key: [t2.c, t2.d], aggs: [count(t.a)] }
+      └─BatchExchange { order: [], dist: HashShard(t2.c, t2.d) }
+        └─BatchHashAgg { group_key: [t2.c, t2.d, t.a], aggs: [] }
```

Comment:
Before this pr, we can not derive the group_key here?

Reply:
Before this pr, we will miss a exchange here.